api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +344,146 @@ type AzureWebhookReceiverConfig struct {
+	// Name is the name of the action to be performed. `Refresh` is the only
+	// action currently supported.
+	//
+	// +kubebuilder:validation:Enum=Refresh;
+	Name GenericWebhookActionName `json:"action" protobuf:"bytes,1,opt,name=action"`
+
+	// MatchExpression defines criteria that a request must meet to trigger this
+	// action.
+	//
+	// +optional
+	MatchExpression string `json:"matchExpression,omitempty" protobuf:"bytes,2,opt,name=matchExpression"`
+
+	// Parameters contains additional parameters for the action.
+	//
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+
+	// Targets is a list of selection criteria for the resources on which the
+	// action should be performed.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Targets []GenericWebhookTarget `json:"targets,omitempty" protobuf:"bytes,4,rep,name=targets"`
+}
+
+// GenericWebhookActionName represents the name of an action to be performed on a resource.
+type GenericWebhookActionName string
+
+const (
+	// GenericWebhookActionNameRefresh indicates a request to refresh the resource.
+	GenericWebhookActionNameRefresh GenericWebhookActionName = "Refresh"
+)
+
+// GenericWebhookTarget describes selection criteria for resources to which some
+// action is to be applied. Name, LabelSelector, and IndexSelector are all optional
+// however, at least one must be specified. When multiple criteria are specified, the
+// results are the combined (logical AND) of the criteria.
+type GenericWebhookTarget struct {
+	// Kind is the kind of the target resource.
+	//
+	// +kubebuilder:validation:Enum=Warehouse;
+	Kind GenericWebhookTargetKind `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+
+	// Name is the name of the target resource. If LabelSelector and/or IndexSelectors
+	// are also specified, the results are the combined (logical AND) of the criteria.
+	//
+	// +optional
+	Name string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+
+	// LabelSelector is a label selector to identify the target resources.
+	// If used with IndexSelector and/or Name, the results are the combined (logical AND) of all the criteria.
+	//
+	// +optional
+	LabelSelector metav1.LabelSelector `json:"labelSelector,omitempty" protobuf:"bytes,3,opt,name=labelSelector"`
+
+	// IndexSelector is a selector used to identify cached target resources by cache key.
+	// If used with LabelSelector and/or Name, the results are the combined (logical AND) of all the criteria.
+	//
+	// +optional
+	IndexSelector IndexSelector `json:"indexSelector,omitempty" protobuf:"bytes,4,opt,name=indexSelector"`
+}
+
+// GenericWebhookTargetKind represents the kind of a target resource.
+type GenericWebhookTargetKind string
+
+const (
+	GenericWebhookTargetKindWarehouse GenericWebhookTargetKind = "Warehouse"
+)
+
+// IndexSelector encapsulates a selector used to derive index keys
+// based on expressions.
```

Comment:
I don't know that mentioning expressions is especially relevant right here. There may not be any expressions involved.

The important thing is that it's mentioned on the `Value` field of the `IndexSelectorRequirement` that the value _could_ be defined using an expression, and you have already got that base covered.